packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -368,6 +368,10 @@ class NewBaseTxPayload(ABC, metaclass=_MetaPayload):
     sender: str
     round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
 
+    @property
+    def data(self) -> Dict[str, Any]:
+        return {k: v for k, v in asdict(self).items() if k not in ["sender", "round_count"]}
+
```

Comment:
It is very confusion that we use the same variable name for different things. For example, we use payload for an instance of `BaseTxPayload`, but also for their content (data).

The use of `allowed_tx_type` and `payload_attribute` makes it hard to understand what information is coming from where. This is exacerbated by inconsistency in naming beyond this, such as through the use of `selection_key` and `collection_key`. 

We have 1 round, 1 payload and transitioning to the next state this payload of the previous round is assumed to be present, hence we should generalize. For now `data` can be used to get the relevant content. In a subsequent PR I suggest to convert this into a payload instance of the expected type for clarity and type-checking reasons. We may set the "sender" to "consensus" (or some special value) to indicate this payload is a consensus payload (merely for the sake of clarity, it is a given since it comes from the synchronized data)

Reply:
`The use of allowed_tx_type and payload_attribute makes it hard to understand what information is coming from where. This is exacerbated by inconsistency in naming beyond this, such as through the use of selection_key and collection_key.` This is really about refactoring the round. We are currently refactoring the payload. KISS